swap_meet/vendor.py

```diff
@@ -1,2 +1,84 @@
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory=None):
+        self.inventory = [] if inventory == None else inventory 
+    
+    def add(self, item):
+        self.inventory.append(item)
+        
+        return item
+    
+    def remove(self, item):
+        result = False
+
+        if item in self.inventory:
+            self.inventory.remove(item)
+            result = item
+
+        return result
+    
+    def get_by_id(self, item_id):
+        results = None
+
+        for item in self.inventory:
+            if item.id == item_id:
+                results = item
+                break
+
+        return results
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        
+        result = False
+
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+                self.remove(my_item)
+                other_vendor.remove(their_item)
+                self.add(their_item)
+                other_vendor.add(my_item)
+                result = True
+        return result
+
+    def swap_first_item(self, other_vendor):
+
+        result = False
+
+        if self.inventory and other_vendor.inventory:
+            self_item = self.inventory[0]
+            self.inventory[0] = other_vendor.inventory[0]
+            other_vendor.inventory[0] = self_item
+            result = True
```

Comment:
Since they tests are necessarily testing for the items to be in the first position after swapping, you could use the `add` and `remove` method as well. However, a considerable advantage you have here is the time complexity of replacing an element at a given index in a list, an `O(1)` operation!